db/models.py

```diff
@@ -50,3 +59,56 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL,
+                             on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return f"{self.created_at}"
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(MovieSession, on_delete=models.CASCADE)
+    order = models.ForeignKey(Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["movie_session", "row", "seat"],
+                                    name="unique_ticket")
+        ]
+
+    def __str__(self) -> str:
+        return (f"{self.movie_session.movie.title} "
+                f"{self.movie_session.show_time} "
+                f"(row: {self.row}, "
+                f"seat: {self.seat})")
+
+    def clean(self) -> None:
+        valid_row = self.movie_session.cinema_hall.rows
+        if self.row not in range(1, valid_row + 1):
```

Comment:
why dont you just check `if not (1 <= self.row <= self.movie_session.cinema_hall.rows)` instead of creating range?

Reply:
i think there is the same speed and memory, so just choose this way to release this  func like this, but now i know that range is not good idea